ably/realtime/connection.py

```diff
@@ -97,6 +117,10 @@ async def ws_read_loop(self):
                 await self.__websocket.close()
                 self.__closed_future.set_result(None)
                 break
+            if action == ProtocolMessageAction.HEARTBEAT:
```

Comment:
We should be checking the ID of the protocol matches the ID we sent before resolving the future